grpc-api/src/main/java/io/deephaven/grpc_api/table/ops/filter/FilterFactory.java

```diff
@@ -0,0 +1,232 @@
+            case EQUALS:
+            case NOT_EQUALS:
+                // At this point, we shouldn't be able to be optimized to a match filter, so we'll tostring and build a condition
+                // and let the DBLangParser turn the "==" into the appropriate java call
+                // Note that case insensitive checks aren't supported on this path
+                if (caseSensitivity != CaseSensitivity.MATCH_CASE) {
+                    throw new IllegalStateException("Should have been compiled out in a previous pass");
+                }
+                return generateConditionFilter(NormalizeFilterUtil.doComparison(operation, caseSensitivity, lhs, rhs));
+            case UNRECOGNIZED:
+            default:
+                throw new IllegalStateException("Can't handle compare operation " + operation);
+        }
+    }
+
+    private SelectFilter generateNumericConditionFilter(CompareCondition.CompareOperation operation, Value lhs, Value rhs) {
+        boolean invert;
+        String columName;
+        Literal value;
+        if (lhs.getDataCase() == Value.DataCase.LITERAL && rhs.getDataCase() == Value.DataCase.REFERENCE) {
+            invert = true;
+            value = lhs.getLiteral();
+            columName = rhs.getReference().getColumnName();
+        } else if (lhs.getDataCase() == Value.DataCase.REFERENCE && rhs.getDataCase() == Value.DataCase.LITERAL) {
+            invert = false;
+            columName = lhs.getReference().getColumnName();
+            value = rhs.getLiteral();
+        } else {
+            // both are references or literals, handle as a condition filter, not range
+            return generateConditionFilter(Condition.newBuilder().setCompare(CompareCondition.newBuilder()
+                    .setOperation(operation)
+                    .setLhs(lhs)
+                    .setRhs(rhs)
+                    .build()).build());
+        }
+        String valueString;
+        switch (value.getValueCase()) {
+            case STRING_VALUE:
+                valueString = value.getStringValue();
+                break;
+            case DOUBLE_VALUE:
+                DecimalFormat format = new DecimalFormat("##0");
+                format.setDecimalSeparatorAlwaysShown(false);
+                format.setGroupingUsed(false);
+                valueString = format.format(value.getDoubleValue());
+                break;
+            case BOOL_VALUE:
+                valueString = Boolean.toString(value.getBoolValue());
+                break;
+            case LONG_VALUE:
+                valueString = Long.toString(value.getLongValue());
+                break;
+            case NANO_TIME_VALUE:
+                valueString = Long.toString(value.getNanoTimeValue());
+                break;
+            case VALUE_NOT_SET:
+            default:
+                throw new IllegalStateException("Range filter can't handle literal type " + value.getValueCase());
```

Comment:
This would be a protocol error instead? Unless the user is actually writing protobuf payloads, this is more of an internal error on the part of the client library?

Reply:
Your point is that it doesn't need to be any friendly for the client-library writer because they will be running a worker that is easily accessible to them? I think if I wrote this code and you made a mistake in the jsapi you'd probably tell me you'd want to see the error msg come back to you without looking at the worker logs (in lieu of the "Internal Error" masked failure).